modules/jdbc/src/main/java/org/apache/ignite/internal/jdbc/JdbcStatement.java

```diff
@@ -414,6 +416,10 @@ public int getUpdateCount() throws SQLException {
 
         JdbcResultSet rs = resSets.get(curRes);
 
+        if (rs == null) {
+            rs = resSets.get(curRes - 1);
+        }
```

Comment:
why do we get the previous result set here?

Reply:
reworked